src/openrct2/interface/Screenshot.cpp

```diff
@@ -339,25 +355,104 @@ static void ReleaseDPI(rct_drawpixelinfo& dpi)
+            rightTileCoords = CoordsXY(gClipSelectionA.x, gClipSelectionB.y);
+            topTileCoords = gClipSelectionA;
+            bottomTileCoords = gClipSelectionB;
+
+            topTileCoords += CoordsXY(16, 16);
+            bottomTileCoords += CoordsXY(16, 16);
+
+            offset = 32;
+        }
+        else if (rotation == 1)
+        {
+            leftTileCoords = gClipSelectionB;
+            rightTileCoords = gClipSelectionA;
+            topTileCoords = CoordsXY(gClipSelectionB.x, gClipSelectionA.y);
+            bottomTileCoords = CoordsXY(gClipSelectionA.x, gClipSelectionB.y);
+
+            leftTileCoords += CoordsXY(32, 32);
+        }
+        else if (rotation == 2)
+        {
+            leftTileCoords = CoordsXY(gClipSelectionA.x, gClipSelectionB.y);
+            rightTileCoords = CoordsXY(gClipSelectionB.x, gClipSelectionA.y);
+            topTileCoords = gClipSelectionB;
+            bottomTileCoords = gClipSelectionA;
+
+            topTileCoords += CoordsXY(16, 16);
+            bottomTileCoords += CoordsXY(16, 16);
+
+            offset = 32;
+        }
+        else
+        {
+            leftTileCoords = gClipSelectionA;
+            rightTileCoords = gClipSelectionB;
+            topTileCoords = CoordsXY(gClipSelectionA.x, gClipSelectionB.y);
+            bottomTileCoords = CoordsXY(gClipSelectionB.x, gClipSelectionA.y);
+
+            rightTileCoords += CoordsXY(32, 32);
+        }
+
+        minY = gClipSelectionA.y / 32;
+        maxY = gClipSelectionB.y / 32;
+        minX = gClipSelectionA.x / 32;
+        maxX = gClipSelectionB.x / 32;
+        int height = GetTallestVisibleTileTopLimited(minX, maxX, minY, maxY, rotation);
+
+        left = translate_3d_to_2d_with_z(rotation, CoordsXYZ(leftTileCoords, 0)).x;
+        right = translate_3d_to_2d_with_z(rotation, CoordsXYZ(rightTileCoords, 0)).x;
+        top = translate_3d_to_2d_with_z(rotation, CoordsXYZ(topTileCoords, height)).y;
+        bottom = translate_3d_to_2d_with_z(rotation, CoordsXYZ(bottomTileCoords, 0)).y;
+    }
+    else
+    {
+        // Run old default code
+        leftTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::LEFT, false);
+        rightTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::RIGHT, false);
+        bottomTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::BOTTOM, false);
+
+        // Centre the coordinates so we don't have a hard crop at the edge of the visible tile
+        leftTileCoords += CoordsXY(16, 16);
+        rightTileCoords += CoordsXY(16, 16);
+        bottomTileCoords += CoordsXY(16, 16);
+
+        // Calculate the viewport bounds
+        left = translate_3d_to_2d_with_z(rotation, CoordsXYZ(leftTileCoords, 0)).x;
+        top = GetTallestVisibleTileTop(mapSize, rotation);
+        right = translate_3d_to_2d_with_z(rotation, CoordsXYZ(rightTileCoords, 0)).x;
+        bottom = translate_3d_to_2d_with_z(rotation, CoordsXYZ(bottomTileCoords, 0)).y;
+    }
```

Comment:
I think this entire function can be reverted. You only have to modify the `GetEdgeTile` function to return the correct tiles for when cut-away view selection is active.

Reply:
Most of the new code has now been migrated to GetEdgeTile. Note that the old code is still on an else branch, otherwise the giant screenshot fails to properly render and the image becomes very large in terms of pixel dimensions.